ouroboros-network-protocols/src/Ouroboros/Network/Protocol/ChainSync/PipeClient.hs

```diff
@@ -0,0 +1,70 @@
+{-# LANGUAGE DataKinds           #-}
+{-# LANGUAGE GADTs               #-}
+{-# LANGUAGE NamedFieldPuns      #-}
+{-# LANGUAGE ScopedTypeVariables #-}
+{-# LANGUAGE RankNTypes #-}
+
+module Ouroboros.Network.Protocol.ChainSync.PipeClient
+where
+
+import Network.TypedProtocol
+import Ouroboros.Network.Protocol.ChainSync.Client
+import Ouroboros.Network.Protocol.ChainSync.Type
+import Pipes
+
+data ChainSyncResponse header point tip
+  = RollForward header tip
+  | RollBackward point tip
+  | IntersectFound point tip
+  | IntersectNotFound tip
+  deriving (Show)
+
+-- | Run a chain sync client as a 'Producer'.
+-- Pass a suitable 'Driver', and a list of known 'point's. The 'Producer' will
+-- initialize itself by requesting an intersection, and yielding either
+-- 'IntersectFound' or 'IntersectNotFound' to indicate the server's response.
+-- After that, it will indefinitely request updates and 'yield' them as
+-- 'RollForward' and 'RollBackward', respectively.
+chainSyncClientProducer :: forall header point tip m dstate
+                         . ( Monad m
+                           )
+                        => Driver
+                            (ChainSync header point tip)
+                            AsClient
+                            dstate
+                            (Producer (ChainSyncResponse header point tip) m)
+                        -> [point]
+                        -> Producer (ChainSyncResponse header point tip) m dstate
+chainSyncClientProducer driver known =
+  snd <$> runPeerWithDriver driver peer
```

Comment:
I have to admit, I'm struggling to understand if this can plausibly work in the way we want it to.

I would have thought that `runPeerWithDriver` will run the peer to the end and not produce anything until it is done, which would not match what we want. My intuition is we'd need to break up `runPeerWithDriver` and integrate it into this function (fortunately it's not long).

But on the other hand, what you have has the right type. So I'm confused :-)

We might need to just try it out.